File: image-colorization/php/image-colorization.php
<!doctype html>
<html>
<head></head>
<body>

<?php
// Basic cURL POST call for OVH colorization API. Will return colorized image.
 
$data = array(
  'source_url' => 'https://www.example.com/image.png',
  'render_factor' => 35
);
 
$payload = json_encode($data);
 
$ch = curl_init('https://api-market-place.ai.ovh.net/image-colorization/process');
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLINFO_HEADER_OUT, true);
curl_setopt($ch, CURLOPT_POST, true);
curl_setopt($ch, CURLOPT_POSTFIELDS, $payload);
curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, 0);
curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, 0);
curl_setopt($ch, CURLOPT_HTTPHEADER, array(
  'Content-Type: application/json',
  'Accept: image/jpg',
  'X-OVH-Api-Key: xxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxx',
  'Content-Length: ' . strlen($payload))
);
   
$response = curl_exec($ch);
$err = curl_error($ch);

curl_close($ch);

if ($err) {
  echo "cURL Error #:" . $err;
} else {
  $fp = fopen('colorized.jpg', 'w');
  fwrite($fp, $response);
  fclose($fp);
  echo '<img src="colorized.jpg" alt="colorized image">';
};
?>
     
</body>
</html>

File: image-colorization/python/image-colorization.py
# Basic cURL POST call for OVH Colorization API. Will return a colorized image located in the script folder.
# Since we will receive non-text code, we will use shutil library to copy object to file
# coding: utf-8

import requests
import shutil

url = "https://api-market-place.ai.ovh.net/image-colorization/process"

payload = {
  'render_factor': 35,
  'source_url': 'https://compress-or-die.com/public/understanding-jpg/assets/improvement_sepia_1.jpg'
}

headers = {
  'X-OVH-Api-Key': "xxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxx",
  'accept': "image/jpg",
  'content-type': "application/json",
  'x-pretty-print': "4"
}

response = requests.request("POST", url, json=payload, headers=headers, stream=True, verify=False)

status = response.status_code

if status == 200:
  with open('colorized.jpg', 'wb') as out_file:
    shutil.copyfileobj(response.raw, out_file)
    del response
    print("Job done !")

else:
  print("Job failed. Error ".response.status_code)
